### DISCUSSION TITLE
button from component

### DISCUSSION THREAD

**Discussion by meavric** [rappasoft/laravel-livewire-tables#1895] [General]
Hi,
I need button to show modal.
Now I use:
`            ->buttons([
                LinkColumn::make('View') 
                    ->title(fn($row) => 'View ' . $row->name)
                    ->location(fn($row) => '#')
                    ->attributes(function($row) {
                        return [
                            'class' => 'btn btn-primary btn-sm',
                            'wire:click.prevent' => 'viewDetailModal('.$row->id.')'
                        ];
                    }),
            ])
`

but I don't want use every time class to define button view, but I want use predefined component <x-button-add>
How can I use it in my situation?
BR

**Comment by lrljoe** [comment]
What do you mean?

If you want to re-use that code, then you could just create a class that extends LinkColumn, with all of that setup by default.

**Comment by meavric** [comment]
Hi,
I know but I use in my project buttons defined by component (<x-button-send>) and I want have in my project one method to manage them.
BR

**Reply by lrljoe** [reply]
You aren't making yourself clear to be honest.

What exactly are you trying to achieve?

Provide some screenshots of what isn't working, or annotate what you're trying to do.

It's perfectly possible to add buttons/views to display Modals from the table.

There are also Action buttons for the top of the table.

You can use Traits to re-use/repurpose specific Columns/Actions/Filters
You can use Extensions to re-use/repurpose specific Columns/Actions/Filters.

Explain exactly (give a full table example) - and explain what you're trying to reuse.

**Comment by meavric** [comment]
Hi
My problem resolved in: https://github.com/rappasoft/laravel-livewire-tables/discussions/1909
BR
